app/scripts/lib/Stores/ExtensionStore.ts

```diff
@@ -0,0 +1,70 @@
+import browser from 'webextension-polyfill';
+import log from 'loglevel';
+import { checkForLastError } from '../../../../shared/modules/browser-runtime.utils';
+import {
+  type IntermediaryStateType,
+  MetaMaskStorageStructure,
+  BaseStore,
+} from './BaseStore';
+
+/**
+ * An implementation of the MetaMask Extension BaseStore system that uses the
+ * browser.storage.local API to persist and retrieve state.
+ */
+export default class ExtensionStore extends BaseStore {
+  isSupported: boolean;
+
+  constructor() {
+    super();
+    this.isSupported = Boolean(browser.storage.local);
+    if (!this.isSupported) {
+      log.error('Storage local API not available.');
+    }
+  }
+
+  /**
+   * Returns all of the keys currently saved
+   *
+   * @returns the key-value map from local storage
+   */
+  async get(): Promise<MetaMaskStorageStructure | null> {
+    if (!this.isSupported) {
+      log.error('Storage local API not available.');
+      return null;
+    }
+    const { local } = browser.storage;
+    const result = await local.get(null);
+    const err = checkForLastError();
+    if (err) {
+      throw err;
+    }
+    return result;
+  }
+
+  /**
+   * Sets the key in local state
+   *
+   * @param obj - The key to set
+   * @param obj.data - The MetaMask State tree
+   * @param obj.meta - The metadata object
+   * @param obj.meta.version - The version of the state tree determined by the
+   * migration
+   * @returns a promise resolving to undefined.
```

Comment:
Nit: Similar to the suggestion for `get`. It's now more specific than "local state", and it has been updated to clarify that all state is overwritten (not just one "key").

Also I've removed `@returns`, since no data is returned. We don't use `@returns` to describe what the type signature tells us, as a general rule. Just to describe what the data _is_ that is returned, which is nothing here.
```suggestion
   * Overwrite data in `local` extension storage area
   *
   * @param obj - The data to set
   * @param obj.data - The MetaMask State tree
   * @param obj.meta - The metadata object
   * @param obj.meta.version - The version of the state tree determined by the
   * migration
```